### ISSUE TITLE
PullRequest clones always default branch

### ISSUE THREAD

**Issue by pawepaw** [ArcBees/teamcity-plugins#47]
Hello,

I'm using teamcity 9.1.4 and I wanted to build my feature branches (all are based on develop branch). I made configuration for VCS root and configured branch specification there:

* vcs root
Default branch:
`develop`

Branch specification:
` +:refs/heads/feature/* `

* I configured pull request plugin like below:

base branch:
`develop`

Teamcity detects pull requests but it builds with sources from develop branch not from feature branches.

**Comment by pawepaw** [comment]
Any info?

**Comment by meriouma** [comment]
Is your branch name of the pull request prefixed by `feature/`? The only time we had this issue is when we didn't write the correct branch specification

**Comment by pawepaw** [comment]
Yes my branch is prefixed by feature.

In teamcity it says it's building correct branch but i spotted this by removing test on the branch. Even after removing, it was running. Checked sources in work directory and it was wrong branch.

**Comment by meriouma** [comment]
Hmm, I'm having a hard time figuring what could be wrong. If TeamCity is building the correct branch, then the trigger works as expected, otherwise you would see a build of `develop` being started. Maybe you can try checking the "Clean all files in the checkout directory before the build" option.

**Comment by pawepaw** [comment]
Checked and it does not help. If i trigger build manually using run button and i chose which branch it all works fine.

**Comment by pawepaw** [comment]
Checked if "Clean all files in the checkout directory before the build" and it does not help.

btw. I've checked logs and teamcity implicitly change branch to develop. I can give you more data to reproduce it if you need.

**Comment by meriouma** [comment]
Thanks for the detailed report. I spotted 3 differences with how we use it :
- We use server side checkout`
- We're not checking "Approve on success"
- We're not putting `-:<default>` in branch specification

Could you try server side checkout?

Regarding versions, we're running the plugin on TeamCity 8.0.1, TeamCity 9.0.1, TeamCity 9.1 and TeamCity 9.1.6

**Comment by pawepaw** [comment]
Tried all except removing "apporve on success". I'll try to apply all of above.

**Comment by Lechus** [comment]
Teamcity 9.0.4 - works with feature branch with settings as @meriouma suggested above.

**Comment by pawepaw** [comment]
any info? Maybe you have any idea what else i can investigate? Tried changing my branch specification from `+:refs/heads/feature/*` to `+:refs/heads/Feature/*` but it makes no difference.

**Comment by meriouma** [comment]
I'm running out of ideas. Can you try watching `+:refs/heads/*` and create a branch without `/feature` to see if it changes anything?

**Comment by pawepaw** [comment]
Yeah, thanks! Started working after changing to `+:refs/heads/*` it works now for every pull request to develop but it's ok.
